From 3ab4a9d75b1ac0f574d59a7850dd96c8fdcebe48 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?N=C3=ADcolas=20Iensen?= <nicolas.iensen@gmail.com>
Date: Fri, 22 Apr 2022 23:22:45 +0000
Subject: [PATCH 1/2] Add --name option to the app generator

In this commit, we are adding the option to set the name of the app
when generating a new one with `rails new`.

The option `--name` will override the application name to be different
from the folder name.

```
rails new my-app-folder --name=my-actual-app-name"
```

The command above will generate a
new Rails application in the folder `my-app-folder`, but the file
`config/application.rb` would have the following structure:

module MyActualAppName
  class Application < Rails::Application
  end
end

This option would be most useful when generating a Rails application in
the current folder:

```
rails new . --name=my-app
```
---
 railties/lib/rails/generators/app_base.rb      | 3 +++
 railties/lib/rails/generators/app_name.rb      | 2 +-
 railties/test/generators/app_generator_test.rb | 5 +++++
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/railties/lib/rails/generators/app_base.rb b/railties/lib/rails/generators/app_base.rb
index 78653fb50d928..4082a797db1d6 100644
--- a/railties/lib/rails/generators/app_base.rb
+++ b/railties/lib/rails/generators/app_base.rb
@@ -24,6 +24,9 @@ def self.strict_args_position
       end
 
       def self.add_shared_options_for(name)
+        class_option :name,                type: :string, aliases: "-n",
+                                           desc: "Name of the app"
+
         class_option :template,            type: :string, aliases: "-m",
                                            desc: "Path to some #{name} template (can be a filesystem path or URL)"
 
diff --git a/railties/lib/rails/generators/app_name.rb b/railties/lib/rails/generators/app_name.rb
index 1ca21bed7840d..a76affca01b3d 100644
--- a/railties/lib/rails/generators/app_name.rb
+++ b/railties/lib/rails/generators/app_name.rb
@@ -11,7 +11,7 @@ def app_name
         end
 
         def original_app_name
-          @original_app_name ||= defined_app_const_base? ? defined_app_name : File.basename(destination_root)
+          @original_app_name ||= defined_app_const_base? ? defined_app_name : (options[:name] || File.basename(destination_root))
         end
 
         def defined_app_name
diff --git a/railties/test/generators/app_generator_test.rb b/railties/test/generators/app_generator_test.rb
index 61d2108f26f5c..bc8aac67d361a 100644
--- a/railties/test/generators/app_generator_test.rb
+++ b/railties/test/generators/app_generator_test.rb
@@ -989,6 +989,11 @@ def test_minimal_rails_app
     assert_no_gem "web-console"
   end
 
+  def test_name_option
+    run_generator [destination_root, "--name=my-app"]
+    assert_file "config/application.rb", /^module MyApp$/
+  end
+
   private
     def stub_rails_application(root = destination_root, &block)
       Rails.application.config.root = root

From fe62bb5762edfc7ee4c3bc3048b77dcb5c4c8136 Mon Sep 17 00:00:00 2001
From: Martin Kalcher <martin.kalcher@aboutsource.net>
Date: Thu, 28 Apr 2022 13:11:19 +0200
Subject: [PATCH 2/2] activerecord: log unnamed bind params

---
 activerecord/lib/active_record/log_subscriber.rb | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/activerecord/lib/active_record/log_subscriber.rb b/activerecord/lib/active_record/log_subscriber.rb
index c45b9dab3e8ac..b5c2942fdfce2 100644
--- a/activerecord/lib/active_record/log_subscriber.rb
+++ b/activerecord/lib/active_record/log_subscriber.rb
@@ -51,7 +51,14 @@ def sql(event)
 
         binds = []
         payload[:binds].each_with_index do |attr, i|
-          attribute_name = attr.respond_to?(:name) ? attr.name : attr[i].name
+          attribute_name = if attr.respond_to?(:name)
+            attr.name
+          elsif attr.respond_to?(:[]) && attr[i].respond_to?(:name)
+            attr[i].name
+          else
+            nil
+          end
+
           filtered_params = filter(attribute_name, casted_params[i])
 
           binds << render_bind(attr, filtered_params)
